src/Agent.Listener/Configuration/ConfigurationManager.cs

```diff
@@ -221,6 +235,30 @@ public AgentSettings LoadSettings()
                 }
             }
 
+            // See if the server supports our OAuth key exchange for credentials
+            if (agent.Authorization != null && 
+                agent.Authorization.ClientId != Guid.Empty &&
+                agent.Authorization.AuthorizationUrl != null)
+            {
+                var credentialData = new CredentialData
+                {
+                    Scheme = Constants.Configuration.OAuth,
+                    Data =
+                    {
+                        { "clientId", agent.Authorization.ClientId.ToString("D") },
+                        { "authorizationUrl", agent.Authorization.AuthorizationUrl.AbsoluteUri },
+                    },
+                };
+
+                // Save the negotiated OAuth credential data
+                _store.SaveCredential(credentialData);
```

Comment:
Out of curiosity, what sprint does the server support this? Is this a server change in sprint 100?